setup/command/pull.go

```diff
@@ -18,13 +19,18 @@ func (f *CommandFactory) Pull() cli.Command {
 				return err
 			}
 
-			provider, err := f.newAWSProviderHelper(c)
+			i := f.NewInstance(args["NAME"])
+			region, err := i.Output(instance.OUTPUT_AWS_REGION)
```

Comment:
Unfortunately, this introduces a 'chicken and egg' problem. A user would need the instance files locally to get the output, but the purpose of this command is to get the instance files locally.  I recommend using a region flag for this command, or update the terraform module + user data script to always specify `us-west-2` when dealing w buckets.

Reply:
It's going to have to be the first option, because when you hard code the terraform module (the s3 bucket resource) to create in us-west-2, then I actually get this error message on the first apply attempt: https://github.com/terraform-providers/terraform-provider-aws/issues/867

When the aws provider and the s3 bucket resource's region don't match, this happens. The bucket gets created but then terraform can't read from it.